sdks/git-sdk/src/main/java/org/jreleaser/sdk/git/ChangelogGenerator.java

```diff
@@ -194,6 +194,11 @@ public Tags resolveTags(Git git, JReleaserContext context) throws GitAPIExceptio
             Version v2 = version(context, tag2, versionPattern);
             return v2.compareTo(v1);
         });
+        
+        GitSdk shallowTest = GitSdk.of(context);
```

Comment:
This check should be performed as the first statement in the `resolveTags` method.